utils/GenerateSpritesheet.java

```diff
@@ -96,7 +190,52 @@ public static void main(String[] args) throws Exception {
 			for (int j = 0; j < sprite_h; ++j)
 			out.setRGB(xpos + i, ypos + j, in.getRGB(i, j));
 		}
-		ImageIO.write(out, "png", new File(result));
+		ImageIO.write(out, "png", result);
 		System.out.println("Success!");
+		if (emitCode) {
+			System.out.println("");
+			ArrayList<String> paths = new ArrayList<>();
+			for (File f = result; !f.getName().equals("sprites"); f = f.getParentFile()) paths.add(0, f.getName());
+			String out_path = "../sprites";
+			for (String f : paths) out_path += "/" + f;
+			String mask_path = new File(out_path).getParent() + "/" + result.getName().substring(0, 2) + "_mask" + result.getName().substring(2);
+
+			for (int f = 0; f < frames_to_render; ++f) System.out.print(
+				"frame_" + f + ":FSET{" +
+					"width_x:" + sprites_x + ";" +
+					"width_y:" + sprites_y + ";" +
+					"tile_width:64;" +
+					"(se_{vert(0.." + (sprites_x - 1) + ")}_{hor(0.." + (sprites_y - 1) + ")}):sheet{" +
+						"x_base:" + f + "*" + sprites_x + "*" + sprite_w + ";y_base:0*" + sprites_y + "*" + sprite_h + "+" + sprite_h + "*2;" +
+						"x_step:" + sprite_w + ";y_step:-" + sprite_h + ";" +
+						"file:\"" + out_path + "\";" +
+						"recolour:\"" + mask_path + "\";" +
+						"x_offset:-32;y_offset:32-" + sprite_h + ";width:" + sprite_w + ";height:" + sprite_h + ";" +
+					"}" +
```

Comment:
I am not sure why you stick so much onto a single line. Whitespace isn't expensive.

Insert a newline character after every line that you have here, and add normal white space. It makes a world of difference in readability with very little extra effort.

It saves a lot of time if someone ever wants or needs to read this data.

Also a diff will be much more readable. In its current form you absolutely will not understand what you changed between versions. With lines that roughly fit without wrapping your chances are much better.

If that breaks copy/pasting, write it to a file instead of dumping it on the screen? Might be useful too if you can split the source file into pieces.